code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -629,3 +629,46 @@
 	message_admins("[ADMIN_LOOKUPFLW(obsessed)] has been made Obsessed by the midround ruleset.")
 	log_game("[key_name(obsessed)] was made Obsessed by the midround ruleset.")
 	return ..()
+
+//////////////////////////////////////////////
+//                                          //
+//            SPIDERS     (GHOST)           //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/spiders
+	name = "Spider Infestation"
+	antag_flag = "Spider"
+	antag_flag_override = ROLE_ALIEN
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_HEAVY
+	required_type = /mob/dead/observer
+	enemy_roles = list("Security Officer", "Detective", "Head of Security", "Captain")
+	required_enemies = list(2,2,1,1,1,0,0,0,0,0)
+	required_candidates = 2
+	weight = 5
+	cost = 10
+	repeatable = TRUE
+	var/list/vents = list()
+
+/datum/dynamic_ruleset/midround/from_ghosts/spiders/execute()
+	for(var/obj/machinery/atmospherics/components/unary/vent_pump/temp_vent in GLOB.machines)
+		if(QDELETED(temp_vent))
+			continue
+		if(is_station_level(temp_vent.loc.z) && !temp_vent.welded)
+			var/datum/pipeline/temp_vent_parent = temp_vent.parents[1]
+			if(!temp_vent_parent)
+				continue // No parent vent
+			if(temp_vent_parent.other_atmosmch.len > 20)
+				vents += temp_vent // Makes sure the pipeline is large enough
+	if(!vents.len)
+		return FALSE
```

Comment:
make this consistent with xenos midround, there's a log message about execute failure. also use length() instead of len

Reply:
Done